src/main/resources/schema/elytron-client-1_1.xsd

```diff
@@ -72,129 +123,401 @@
+        </xsd:annotation>
         <xsd:all minOccurs="0">
-            <xsd:element name="global" type="empty-type" />
-            <xsd:element name="use-service-loader" type="module-ref-type" />
+            <xsd:element name="global" type="empty-type" >
+                <xsd:annotation>
+                    <xsd:documentation>
+                        The providers from java.security.Security.getProviders()
+                    </xsd:documentation>
+                </xsd:annotation>
+            </xsd:element>
+            <xsd:element name="use-service-loader" type="module-ref-type" >
+                <xsd:annotation>
+                    <xsd:documentation>
+                        Providers loaded using service loader discovery from the module specified, 
+                        if no module is specified the ClassLoader which loaded the authentication client is used.
+                    </xsd:documentation>
+                </xsd:annotation>
+            </xsd:element>
         </xsd:all>
     </xsd:complexType>
 
     <xsd:element name="abstract-sasl-factories" abstract="true"/>
-    <xsd:element name="use-provider-sasl-factory" substitutionGroup="abstract-sasl-factories" type="empty-type" />
-    <xsd:element name="use-service-loader-sasl-factory" substitutionGroup="abstract-sasl-factories" type="module-ref-type" />
+    <xsd:element name="use-provider-sasl-factory" substitutionGroup="abstract-sasl-factories" type="empty-type" >
+        <xsd:annotation>
+            <xsd:documentation>
+                The java.security.Provider instances either inherited or defined in this configuration will be used to locate the available SASL client factories.
+            </xsd:documentation>
+        </xsd:annotation>
+    </xsd:element>
+    <xsd:element name="use-service-loader-sasl-factory" substitutionGroup="abstract-sasl-factories" type="module-ref-type" >
+        <xsd:annotation>
+            <xsd:documentation>
+                SASL client factories will be discovered using service loader discovery on the specified module or if not specified using the ClassLoader loading the configuration.
+            </xsd:documentation>
+        </xsd:annotation>
+    </xsd:element>
 
     <xsd:complexType name="certificate-revocation-list-type">
-        <xsd:attribute name="path" type="xsd:string" use="optional"/>
+        <xsd:annotation>
+            <xsd:documentation>
+                The presence of this element enabled checking the peer's certificate against a certificate revocation list.
+            </xsd:documentation>
+        </xsd:annotation>
+        <xsd:attribute name="path" type="xsd:string" use="optional">
+            <xsd:annotation>
+                <xsd:documentation>
+                    Path to the certificate revocation list.
+                </xsd:documentation>
+            </xsd:annotation>
+        </xsd:attribute>
         <xsd:attribute name="resource" type="resource-type" use="optional"/>
```

Comment:
The module resource to use as certificate revocation list. Alternative to "path" and "uri".